tableauserverclient/models/schedule_item.py

```diff
@@ -0,0 +1,203 @@
+        self.schedule_type = schedule_type
+
+    @property
+    def created_at(self):
+        return self._created_at
+
+    @property
+    def end_schedule_at(self):
+        return self._end_schedule_at
+
+    @property
+    def execution_order(self):
+        return self._execution_order
+
+    @execution_order.setter
+    def execution_order(self, value):
+        if value and not hasattr(ScheduleItem.ExecutionOrder, value):
+            error = "Invalid execution order defined: {}.".format(value)
+            raise ValueError(error)
+        else:
+            self._execution_order = value
+
+    @property
+    def frequency(self):
+        return self._frequency
+
+    @property
+    def id(self):
+        return self._id
+
+    @property
+    def name(self):
+        return self._name
+
+    @name.setter
+    def name(self, value):
+        if not value:
+            error = "Name must be defined."
+            raise ValueError(error)
+        else:
+            self._name = value
+
+    @property
+    def next_run_at(self):
+        return self._next_run_at
+
+    @property
+    def priority(self):
+        return self._priority
+
+    @priority.setter
+    def priority(self, value):
+        if value < 1 or value > 100:
+            error = "Invalid priority defined: {}.".format(value)
+            raise ValueError(error)
+        else:
+            self._priority = value
+
+    @property
+    def schedule_type(self):
+        return self._schedule_type
+
+    @schedule_type.setter
+    def schedule_type(self, value):
+        if not value:
+            error = "Schedule type must be defined."
+            raise ValueError(error)
+        elif not hasattr(ScheduleItem.Type, value):
+            error = "Invalid schedule type defined: {}.".format(value)
+            raise ValueError(error)
+        else:
+            self._schedule_type = value
+
+    @property
+    def state(self):
+        return self._state
+
+    @state.setter
+    def state(self, value):
+        if not hasattr(ScheduleItem.State, value):
+            error = "Invalid state defined."
+            raise ValueError(error)
+        else:
+            self._state = value
+
+    @property
+    def updated_at(self):
+        return self._updated_at
+
+    def _parse_common_tags(self, schedule_xml):
+        if not isinstance(schedule_xml, ET.Element):
+            schedule_xml = ET.fromstring(schedule_xml).find('.//t:schedule', namespaces=NAMESPACE)
+        if schedule_xml is not None:
+            (_, name, _, _, updated_at, _, frequency, next_run_at, end_schedule_at, execution_order,
+             priority, interval_item) = self._parse_element(schedule_xml)
+
+            self._set_values(None, name, None, None, updated_at, None, frequency, next_run_at, end_schedule_at,
+                             execution_order, priority, interval_item)
+
+        return self
+
+    def _set_values(self, id, name, state, created_at, updated_at, schedule_type, frequency,
```

Comment:
@RussTheAerialist I took a local copy of this PR to try and address my own feedback and put it up for review, and this method made me wonder if there was a better way.

I'm thinking of instead of having a giant setter that does nothing but check that attributes are not false -- what if something like this:

1. The ScheduleItem constructor has everything initialized to None (or has default args)
2. _parse_element returns only non-null/None attributes it gets back as a dict
3. We can loop over the dict to apply the attributes in _set_values (and remove the giant function signature) or I think we can do away with the method entirely, but the idea was on the tip of my tongue and it left me.

I can dig in farther, but for now I've at least done this:
```
            schedule_item._set_values(id=id,
                                      name=None,
                                      state=state,
                                      created_at=created_at,
                                      updated_at=updated_at,
                                      schedule_type=None,
                                      frequency=frequency,
                                      next_run_at=next_run_at,
                                      end_schedule_at=end_schedule_at,
                                      execution_order=None,
                                      priority=None,
                                      interval_item=None)
```

Thoughts? (I can do that in a separate PR)